src/lib/libwasi.js

```diff
@@ -262,6 +262,10 @@ var WasiLibrary = {
     if (printCharBuffers[1].length) printChar(1, {{{ charCode("\n") }}});
     if (printCharBuffers[2].length) printChar(2, {{{ charCode("\n") }}});
   },
+  // When we don't have a filesystem enabled, we do not need to proxy fd_write
+  // to the main thread, because all we do is log out() and err() calls.
+  // This enables printf() to work from Wasm Workers.
+  fd_write__proxy: 'none',
```

Comment:
Can you make this a separate PR?    It seems like a good idea but unrelated to the `abort` stuff.

Reply:
Figuring out exactly what is going on here seems like a good reason to split out the fdwrite change... but i get that you might be tired on splitting up changes at this point :)  

Personally I'd love to see the assertions fix + fdwrite fix land first, then the new modes.   But I really appreciate all the PRs you've been landing recently and I don't want to discourage your too much so lgtm either way.

Reply:
Ok, I'll defer this feature altogether now for the time being, to avoid any chances.